src/widget.rs

```diff
@@ -34,12 +44,41 @@ pub trait WidgetExtManual {
 
     fn override_symbolic_color(&self, name: &str, color: &RGBA);
 
+    fn connect_drag_data_get<F: Fn(&Self, &DragContext, &mut SelectionData, u32, u32) + 'static>(&self, f: F) -> u64;
+
     fn connect_map_event<F: Fn(&Self, &Event) -> Inhibit + 'static>(&self, f: F) -> u64;
 
     fn connect_unmap_event<F: Fn(&Self, &Event) -> Inhibit + 'static>(&self, f: F) -> u64;
 }
 
 impl<O: IsA<Widget> + IsA<Object>> WidgetExtManual for O {
+
+    fn drag_dest_set(&self, flags: DestDefaults, targets: &[TargetEntry], actions: DragAction) {
+        let stashes: Vec<_> = targets.iter().map(|e| e.to_glib_none()).collect();
+        let mut t = Vec::with_capacity(stashes.len());
+        for stash in stashes {
+            t.push(stash.0.clone());
+        }
+        unsafe { ffi::gtk_drag_dest_set(self.to_glib_none().0,
+                                        flags.to_glib(),
+                                        t.as_mut_ptr() as *mut ffi::GtkTargetEntry,
+                                        t.len() as i32,
+                                        actions.to_glib())};
+    }
+
+    fn drag_source_set(&self, start_button_mask: ModifierType, targets: &[TargetEntry], actions: DragAction) {
+        let stashes: Vec<_> = targets.iter().map(|e| e.to_glib_none()).collect();
+        let mut t = Vec::with_capacity(stashes.len());
+        for stash in stashes {
+            t.push(stash.0.clone());
```

Comment:
It's just cloning the pointer I realized, so this is definitely wrong. I've pushed an updated version that copies all the `TargetEntry`s into new `ffi::GdkTargetEntry`s that I think is more correct, though I'm not certain if it's right. But it removes the null pointer problem revealed in gtk-rs/examples#116